### PR TITLE
[Backport 2.3] #12625: Add Current Date to update_time Field for Block and Pages

### PR DESCRIPTION
**Pull request by osrecio**
Set Current Date Time to update_time field

### Description
add current date time to 'update_time' in `_beforeSave` method  for Page and Block

### Fixed Issues (if relevant)
1. magento/magento2#12625: when saving a page in magento 2.2.1, 'Modified' date field is not getting updated

### Manual testing scenarios
1. Login to magento admin
2. Go to Content > Pages
3. In Action column, click select > edit
4. Make changes to that page and click 'Save Page'

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [X] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by dmanners** [comment]
This is a backport of https://github.com/magento/magento2/pull/12636

File: app/code/Magento/Cms/Model/ResourceModel/Page.php [hunk 1]
```diff
@@ -126,6 +126,10 @@ protected function _beforeSave(AbstractModel $object)
                 __('The page URL key cannot be made of only numbers.')
             );
         }
+
+        $date = (new \DateTime())->format(DateTime::DATETIME_PHP_FORMAT);
```
Comment (sidolov) [inline app/code/Magento/Cms/Model/ResourceModel/Page.php]: Please, use framework implementation for date value \Magento\Framework\Stdlib\DateTime\DateTime::date

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by osrecio** [comment]
@sidolov Changes made, and added integration tests for Cms Block 😃

**Comment by sidolov** [comment]
Hi @osrecio , i investigated this issue deeper and find that field 'update_time' should be automatically updated by Mysql engine. In db this field has type timestamp and must be automatically updated after row has been changed. We had the same issue with quote (bf9ea28eef07046fabb78dcde84951382b7f0716). 
This behavior fixing by unset 'update_time' field before model save: 

```
if ($this->hasDataChanges()) {
    $this->unsUpdateTime();
}
```

**Comment by osrecio** [comment]
Hi @sidolov I changed Models to add in method `beforeSave`:

```
if ($this->hasDataChanges()) {
    $this->unsUpdateTime();
}
```
Thanks for catch it, I made my life complicated changing Resource Model :)

**Review by sidolov — APPROVED** [review]

**Pull request merged**